pkg/crypto/helpers.go

```diff
@@ -121,7 +122,12 @@ func TopicFromCert(cert *x509.Certificate) (string, error) {
 	var oidASN1UserID = asn1.ObjectIdentifier{0, 9, 2342, 19200300, 100, 1, 1}
 	for _, v := range cert.Subject.Names {
 		if v.Type.Equal(oidASN1UserID) {
-			return v.Value.(string), nil
+			var uid = v.Value.(string)
+			if strings.HasPrefix(uid, "com.apple.mgmt") {
+				return v.Value.(string), nil
+			} else {
```

Comment:
there's no need for an `else` branch if you have a return.

Reply:
I'm not sure I understand you correctly on this one. If no UID is found, the function returns an error. If the UID is present, then the code will check if it has the required prefix or not. In either case it should break the loop, which the return does, but in one case it should return the UID value, in the other case it should return an error. 

If I remove the else block the method will return the error message at the bottom of the method instead a more explanatory error message defined in the else block.

Reply:
> If no UID is found, the function returns an error

and function execution stops because the error was returned. The behavior is the same wether `else` is present or not. 
All I'm advocating for is to remove the extra branch and keep the code from within the `else` statement at a different indentation level. 

This comment would also come from `golint` I believe. 

Take this example: 
```
func get(i int) int {
    if f := i * 2; f == 6 {
        return f
    } else {
        return f * 3
    }
}
``` 

The code is correct, but the recommended go style is to write it like this:

```
func get(i int) int {
    f := i * 2
    if f == 6 {
        return f
    }
    return f * 3
}
```